src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -310,6 +310,23 @@ impl KvmVcpu {
             }
             _ => (),
         }
+
+        // If a custom CPU template is configured, append MSRs from its CPU configuration
+        // for setting them to KVM.
+        if let CpuConfigurationType::Custom(cpu_template) = &vcpu_config.cpu_config {
+            let entries: Vec<kvm_msr_entry> = cpu_template
+                .msrs
+                .iter()
+                .map(|(&addr, &value)| kvm_msr_entry {
+                    index: addr,
+                    data: value,
+                    ..Default::default()
+                })
+                .collect();
+
+            msr_boot_entries.extend(entries);
```

Comment:
IIUC, we also need to extend `self.msr_list` that is a list of MSR indices to save into snapshots. Otherwise, snapshot restore may fail.

Reply:
Good point, thanks! Updated.